linera-faucet/server/src/lib.rs

```diff
@@ -150,16 +153,32 @@ where
             .open_chain(ownership, ApplicationPermissions::default(), self.amount)
             .await;
         self.context.lock().await.update_wallet(&client).await?;
-        let (message_id, certificate) = match result? {
-            ClientOutcome::Committed(result) => result,
-            ClientOutcome::WaitForTimeout(timeout) => {
-                return Err(Error::new(format!(
-                    "This faucet is using a multi-owner chain and is not the leader right now. \
-                    Try again at {}",
-                    timeout.timestamp,
-                )));
-            }
-        };
+        let (message_id, certificate) = result?.try_unwrap()?;
+
+        if client.next_block_height() >= self.end_block_height {
+            let key_pair = client.key_pair().await?;
+            let balance = client.local_balance().await?.try_sub(Amount::ONE)?;
+            let ownership = client.chain_state_view().await?.ownership().clone();
+            let (message_id, certificate) = client
+                .open_chain(ownership, ApplicationPermissions::default(), balance)
+                .await?
+                .try_unwrap()?;
+            // TODO(#1795): Move the remaining tokens to the new chain.
+            client.close_chain().await?.try_unwrap()?;
+            let chain_id = ChainId::child(message_id);
+            info!("Switching to a new faucet chain {chain_id:8}; remaining balance: {balance}");
+            self.context
+                .lock()
+                .await
+                .update_wallet_for_new_chain(
+                    chain_id,
+                    Some(key_pair),
```

Comment:
We use the same keypair on child chains?

Reply:
It was easiest to do it that way, and maybe it's also better: That way you don't have to keep backing up new key pairs.

Reply:
This will change in a moment with the new `Signer` abstraction anyway.